applications: asset_tracker_v2: Add air quality (IAQ) readings (#6941)
nrfconnect/sdk-nrf: applications/asset_tracker_v2/doc/sensor_module.rst

   This is beyond the scope of the general asset tracker framework this application provides.
   Therefore, the readings are not transmitted to the cloud and are only used to detect a binary active and inactive state.

.. _bosch_software_environmental_cluster_library:

Bosch Software Environmental Cluster (BSEC) library
===================================================

The sensor module supports integration with the BSEC signal processing library using the external sensors, internal convenience API.
If enabled, the BSEC library is used instead of the BME680 Zephyr driver to provide sensor readings from the BME680 for temperature, humidity, and atmospheric pressure.
In addition, the BSEC driver provides an additional sensor reading, indoor air quality (IAQ), which is a metric given in between 0-500 range, that estimates the air quality of the environment.

> As the BSEC library requires a separate license, it is not a default part of NCS, but can be downloaded externally and imported into the NCS source tree.

Reviewer: ```suggestion
As the BSEC library requires a separate license, it is not a default part of |NCS|, but can be downloaded externally and imported into the |NCS| source tree.
```

   This is beyond the scope of the general asset tracker framework this application provides.
   Therefore, the readings are not transmitted to the cloud and are only used to detect a binary active and inactive state.

Bosch Software Environmental Cluster (BSEC) library
===================================================

The sensor module supports integration with the Bosch BSEC signal processing library via the *external sensors* internal convenience API.
If enabled, the BSEC library is used instead of the BME680 Zephyr driver to provide sensor readings from the BME680 for temperature, humidity and atmospheric pressure.

> In addition, the BSEC driver provides an additional sensor reading, indoor air quality (IAQ), which is a metric given in the range of 0-500 that estimates the air quality of the
> environment.

Reviewer 2: ```suggestion
In addition, the BSEC driver provides an additional sensor reading, indoor air quality (IAQ), which is a metric given in the range of 0-500 that estimates the air quality of the environment.
```

Due to the BSEC library requiring separate lisencing, it is not a default part of NCS, but can be downloaded externally and imported into the NCS source tree.
To enable Bosch BSEC perform the following steps:

1. Download the Bosch BSEC library, available through the following link `Bosch BSEC`_.
#. Place the content of the library in the path given by `CONFIG_EXTERNAL_SENSORS_BME680_BSEC_PATH` or update the path configuration to reference the location of the library.
#. Disable the Zephyr BME680 driver by setting :kconfig:`CONFIG_BME680` to false.

> #. Enable the *external sensors* API BSEC integration layer by enabling `CONFIG_EXTERNAL_SENSORS_BME680_BSEC`.

Reviewer 2: **question:** Why not :kconfig:`CONFIG_EXTERNAL_SENSORS_BME680_BSEC` here?

Author: I believe this is because the configuration resides in the application and not as a sub-library in NCS. @divipillai might have a better answer than me. But, this is typically how it is done when referencing app-specific kconfigs.

Reviewer: Yes, we mention :kconfig: option when we have to refer to the configuration options in the kconfig-library in NCS https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/kconfig/index.html. The application-specific configuration options are listed in the "Configuration options" section and here we mention as :ref:`CONFIG_EXTERNAL_SENSORS_BME680_BSEC <CONFIG_EXTERNAL_SENSORS_BME680_BSEC_PATH>` to link it to the corresponding configuration options.

I will be making these changes in the PR.

Air quality readings will be provided with the :c:enum`SENSOR_EVT_ENVIRONMENTAL_DATA_READY` event.

External sensors API BSEC configurations
----------------------------------------

.. _CONFIG_EXTERNAL_SENSORS_BME680_BSEC

CONFIG_EXTERNAL_SENSORS_BME680_BSEC - Use Bosch BSEC library for the BME680.

.. _CONFIG_EXTERNAL_SENSORS_BME680_BSEC_PATH

CONFIG_EXTERNAL_SENSORS_BME680_BSEC_PATH - Path of the Bosch BSEC library folder.

.. _CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_RATE_ULTRA_LOW_POWER

CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_RATE_ULTRA_LOW_POWER - BSEC ultra low power sample rate.

.. _CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_RATE_LOW_POWER

CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_RATE_LOW_POWER - BSEC low power sample rate.

> .. _CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_RATE_HIGH_POWER
> 
> CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_RATE_HIGH_POWER - BSEC ultra high power sample rate.

Reviewer 2: ```suggestion
.. _CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_RATE_HZ_CONTINUOUS

CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_RATE_HZ_CONTINUOUS - BSEC sample rate in case of Continuous Mode in Hertz (Hz).
```

Reviewer 2: Third mode is named `continuous`.

> .. _CONFIG_EXTERNAL_SENSORS_BSEC_TEMPERATURE_OFFSET
> 
> CONFIG_EXTERNAL_SENSORS_BSEC_TEMPERATURE_OFFSET - BSEC temperature offset in degree celsius * 100.

Reviewer 2: ```suggestion
.. _CONFIG_EXTERNAL_SENSORS_BSEC_TEMPERATURE_CELSIUS_OFFSET

CONFIG_EXTERNAL_SENSORS_BSEC_TEMPERATURE_CELSIUS_OFFSET - BSEC temperature offset in degree celsius * 100.
```
